ios/FluentUI/Controls/IndeterminateProgressBarView.swift

```diff
@@ -0,0 +1,143 @@
+        super.layoutSubviews()
+
+        trackLayer.removeFromSuperlayer()
+        trackLayer.frame = CGRect(x: 0, y: 0, width: self.frame.width, height: self.frame.size.height)
+        layer.addSublayer(trackLayer)
+
+        addGradientLayer()
+        stopAnimating()
+        animationGroup = addAnimation()
+        startAnimating()
+    }
+
+    open override func didMoveToWindow() {
+        super.didMoveToWindow()
+        updateColors()
+    }
+
+    @objc open func startAnimating() {
+        guard !isAnimating else {
+            return
+        }
+        isAnimating = true
+
+        isHidden = false
+        gradientLayer.add(animationGroup, forKey: Constants.animationKey)
+    }
+
+    @objc open func stopAnimating() {
+        guard isAnimating else {
+            return
+        }
+        isAnimating = false
+
+        if hidesWhenStopped {
+            isHidden = true
+        }
+        gradientLayer.removeAnimation(forKey: Constants.animationKey)
+    }
+
+    public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        if #available(iOS 13, *) {
+            if previousTraitCollection?.userInterfaceStyle != traitCollection.userInterfaceStyle {
+                updateColors()
+            }
+        }
+    }
+
+    private func addGradientLayer() {
+        gradientLayer.removeFromSuperlayer()
+        gradientLayer.frame = CGRect(x: 0, y: 0, width: frame.width / 2, height: frame.height)
+        gradientLayer.startPoint = CGPoint(x: 0.0, y: 1.0)
+        gradientLayer.endPoint = CGPoint(x: 1.0, y: 1.0)
+        gradientLayer.locations = [0.0, 0.5, 1.0]
+        layer.addSublayer(gradientLayer)
+    }
+
+    private func addAnimation() -> CAAnimationGroup {
+        let groupAnimation = CAAnimationGroup()
+
+        let position = CABasicAnimation(keyPath: "position.x")
+        position.fromValue = -frame.width / 2
+        position.toValue = frame.width * 1.5
+
+        let opacity = CAKeyframeAnimation(keyPath: "opacity")
+        opacity.keyTimes = Constants.opacityKeyFrame
+        opacity.values = Constants.opacityValues
+        opacity.isRemovedOnCompletion = false
+        opacity.fillMode = CAMediaTimingFillMode.forwards
+
+        groupAnimation.animations = [opacity, position]
+        groupAnimation.repeatCount = .infinity
+        groupAnimation.duration = Constants.animationDuration
+        groupAnimation.timingFunction = Constants.animationTimingFunction
+        return groupAnimation
+    }
+
+    private func updateColors() {
+        if let window = window {
+            let appColor = Colors.primary(for: window)
+            let edgeColor = appColor.withAlphaComponent(0).cgColor
+            let centerColor = appColor.cgColor
+
+            gradientLayer.colors = [edgeColor, centerColor, edgeColor]
+            trackLayer.backgroundColor = Colors.Progress.trackTint.cgColor
```

Comment:
If this doesn't change per window, just set it in the init and don't change later?

Reply:
No, the trackLayer's backgroundColor needs to be updated when switching between light mode and dark mode.